### DISCUSSION TITLE
箇条書きのマージンについて

### DISCUSSION THREAD

**Discussion by yoshi389111** [increments/qiita-discussions#420] [要望・アイデア・不具合]
箇条書きをした場合に、行間が間延びした表示になることがあります。

以下の画像は、どちらも最近（しばらく前のマークダウンエンジンの変更後に）新規作成or編集している記事なのですが、箇条書きの行間が異なります。

間延びした表示の方（左側）は、 `<li>` の中に `<p>` があって、`.it-MdContent ul p` というセレクタにヒットしてマージンが適用されているようです（スクリーンショットは適用されているマージンをハイライトしている状態です）。

もう片方の間延びしていない方（右側）には `<li>` の中には `<p>` がないです。

![image](https://user-images.githubusercontent.com/15214393/214221361-1fe0e6ff-57df-4088-8ca7-7d6ddecf7adc.png)

これは何が違うのでしょうか？

環境は、Windows11 + Chrome 109.0.5414.75 です。

該当の記事は [これ(左側)](https://qiita.com/yoshi389111/items/53ca1a7a65ca9814b27c) と [これ(右側)](https://qiita.com/yoshi389111/items/6ba5966f739ea01c6719) です。

**Comment by tanaka-asami415** [comment]
@yoshi389111 
ご報告ありがとうございます🙏
箇条書きの一部に空行が入ってしまうと、箇条書き全体の行間が間延びしてしまうようです。

修正が可能かも含めて調査いたしますが、空行を詰めていただくことで正しい行間になりますので、
暫定的な対応として空行の削除等をお試しいただけますと幸いです🙇

```
現状

* foo
* bar

* baz
* quz

空行をつめる

* foo
* bar
* baz
* qux

もしくはこのように別の文章や罫線を追加

* foo
* bar

---

* baz
* quz
```

**Reply by yoshi389111** [reply]
了解しました。ご確認ありがとうございました。

**Reply by yarakos95** [reply]
横から失礼します。
@tanaka-asami415 さま

> 修正が可能かも含めて調査いたしますが、空行を詰めていただくことで正しい行間になりますので、暫定的な対応として空行の削除等をお試しいただけますと幸いです🙇

現在、Qiita Markdown は CommonMark に準拠しています。本 Discussion に取り上げられているリストの挙動は CommonMark 仕様として記載されています。これが独自に変更される場合、CommonMark ではない別の Markdown 仕様となってしまいます。

>> A list is [loose](https://spec.commonmark.org/0.30/#loose) if any of its constituent list items are separated by blank lines, or if any of its constituent list items directly contain two block-level elements with a blank line between them. Otherwise a list is [tight](https://spec.commonmark.org/0.30/#tight). (The difference in HTML output is that paragraphs in a loose list are wrapped in `<p>` tags, while paragraphs in a tight list are not.)
>
> [CommonMark Spec 0.30](https://spec.commonmark.org/0.30/)

私はそのような変更を加えることに反対です。Qiita Markdown で CommonMark にない新たな拡張が加わることは歓迎しますが、CommonMark にある仕様そのものを変更することには反対です。もしもこの挙動を変更する場合、これは「修正」ではなく「独自仕様への変更」に当たると思います。

**Reply by tanaka-asami415** [reply]
@yoshi389111  @Yarakashi-Kikohshi 

Yarakashi-Kikohshi さん。ありがとうございます！

社内で議論した結果、拡張は行わない方針となりました。
空行を詰めていただく等で対応いただければと思います。
お手数をおかけしますが、よろしくお願いします🙇

**Reply by yoshi389111** [reply]
本件、了解です。お手数をおかけしました。

[CommonMark Spec 0.30](https://spec.commonmark.org/0.30/) を見ると、箇条書きのマーカー文字を変えると（空行を開けても、開けなくても）タイトな別リストになるようなので、それで対応することします。

```
- foo
- bar
+ baz
```

あるいは

```
- foo
- bar

+ baz
```

ありがとうございました。
